ibm/resource_ibm_is_instance_network_interface.go

```diff
@@ -0,0 +1,649 @@
+			ValidateFunctionIdentifier: ValidateRegexpLen,
+			Type:                       TypeString,
+			Optional:                   true,
+			Regexp:                     `^([a-z]|[a-z][-a-z0-9]*[a-z0-9])$`,
+			MinValueLength:             1,
+			MaxValueLength:             63,
+		},
+		ValidateSchema{
+			Identifier:                 isInstanceNicPrimaryIpv4Address,
+			ValidateFunctionIdentifier: ValidateRegexp,
+			Type:                       TypeString,
+			Optional:                   true,
+			Regexp:                     `^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$`,
+		},
+	)
+
+	resourceValidator := ResourceValidator{ResourceName: "ibm_is_instance_network_interface", Schema: validateSchema}
+	return &resourceValidator
+}
+
+func resourceIBMIsInstanceNetworkInterfaceCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vpcClient, err := meta.(ClientSession).VpcV1API()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	instance_id := d.Get("instance").(string)
+
+	createInstanceNetworkInterfaceOptions := &vpcv1.CreateInstanceNetworkInterfaceOptions{}
+
+	createInstanceNetworkInterfaceOptions.SetInstanceID(instance_id)
+
+	subnetId := d.Get(isInstanceNicSubnet).(string)
+	subnetIdentity := vpcv1.SubnetIdentity{
+		ID: &subnetId,
+	}
+	createInstanceNetworkInterfaceOptions.SetSubnet(&subnetIdentity)
+	if allow_ip_spoofing, ok := d.GetOk(isInstanceNicAllowIPSpoofing); ok {
+		createInstanceNetworkInterfaceOptions.SetAllowIPSpoofing(allow_ip_spoofing.(bool))
+	}
+	if name, ok := d.GetOk(isInstanceNicName); ok {
+		createInstanceNetworkInterfaceOptions.SetName(name.(string))
+	}
+	if primary_ipv4, ok := d.GetOk(isInstanceNicPrimaryIpv4Address); ok {
+		createInstanceNetworkInterfaceOptions.SetPrimaryIpv4Address(primary_ipv4.(string))
+	}
+
+	if secgrpintf, ok := d.GetOk(isInstanceNicSecurityGroups); ok {
+		secgrpSet := secgrpintf.(*schema.Set)
+		if secgrpSet.Len() != 0 {
+			var secgrpobjs = make([]vpcv1.SecurityGroupIdentityIntf, secgrpSet.Len())
+			for i, secgrpIntf := range secgrpSet.List() {
+				secgrpIntfstr := secgrpIntf.(string)
+				secgrpobjs[i] = &vpcv1.SecurityGroupIdentity{
+					ID: &secgrpIntfstr,
+				}
+			}
+			createInstanceNetworkInterfaceOptions.SecurityGroups = secgrpobjs
+		}
+	}
+
+	networkInterface, response, err := vpcClient.CreateInstanceNetworkInterfaceWithContext(context, createInstanceNetworkInterfaceOptions)
+	if err != nil {
+		log.Printf("[DEBUG] CreateInstanceNetworkInterfaceWithContext failed %s\n%s", err, response)
+		return diag.FromErr(fmt.Errorf("CreateInstanceNetworkInterfaceWithContext failed %s\n%s", err, response))
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", *createInstanceNetworkInterfaceOptions.InstanceID, *networkInterface.ID))
+	d.Set("network_interface", *networkInterface.ID)
+
+	_, err = isWaitForNetworkInterfaceAvailable(vpcClient, d.Id(), d.Timeout(schema.TimeoutUpdate), d)
```

Comment:
Isnt this redundant?
line number 266 is anyway waiting..

Reply:
first one is to wait for networkintf after creating networkintf. second wait is to wait after floatingip is added to the networkintf. i have moved the second wait call inside the floatingip attachment IF loop